src/commands/log/mod.ts

```diff
@@ -149,6 +154,68 @@ function fmtSession(idx: string, sess: Session, indentWidth: number) {
     return strings.join("");
 }
 
+function printTimesheet(sessions: Session[]) {
+    // bucket sessions into days
+
+    let buckets: Map<number, number> = new Map();
+    let table: string[][] = [];
+    sessions.forEach((session) => {
+        let start = new Date(session.start);
+        let end = new Date(session.end ? session.end : Date.now());
+        const [startyear, startmonth, startdate] = destructureDate(start);
+        const [endyear, endmonth, enddate] = destructureDate(end);
+
+        const updateBucket = (key: number, val: number) => {
+            let prevVal = buckets.get(key) || 0;
+            buckets.set(key, Math.min(prevVal + val, 24 * 60 * 60 * 1000));
+        };
+
+        if (startdate === enddate) {
+            // session does not spill into another day
+            const key = new Date(startyear, startmonth - 1, startdate).getTime();
+            let hours = end.getTime() - start.getTime();
+            updateBucket(key, hours);
```

Comment:
Would it be possible to do an early return here to avoid the big else block?

Reply:
it keeps the logic explicit and doesn't incur any performance loss so idt it needs to be changed